functionMap.php

```diff
@@ -13,38 +13,37 @@
 /**
  * This array is in the same format as the function map array in PHPStan:
  *
- * '<function_name>' => ['<return_type>', '<arg_name>'=>'<arg_type>']
+ * '<function_name>' => ['<return_type>', '<arg_name>' => '<arg_type>']
  *
  * For classes, or if you don't wish to define the `@phpstan-return` tag:
  *
- * '<class_name>' => [null, '<arg_name>'=>'<arg_type>']
+ * '<class_name>' => [null, '<arg_name>' => '<arg_type>']
  *
- * @link https://github.com/phpstan/phpstan-src/blob/1.5.x/resources/functionMap.php
+ * @link https://github.com/phpstan/phpstan-src/blob/1.10.x/resources/functionMap.php
  */
 return [
-    'addslashes_gpc' => ['T', '@phpstan-template'=>'T', 'gpc'=>'T'],
-    'get_objects_in_term' => [null, 'args'=>'array{order?: string}'],
-    'have_posts' => [null, '@phpstan-impure'=>''],
-    'rawurlencode_deep' => ['T', '@phpstan-template'=>'T', 'value'=>'T'],
-    'sanitize_category' => ['T', '@phpstan-template'=>'T of array|object', 'category'=>'T'],
-    'sanitize_post' => ['T', '@phpstan-template'=>'T of array|object', 'post'=>'T'],
-    'sanitize_term' => ['T', '@phpstan-template'=>'T of array|object', 'term'=>'T'],
-    'stripslashes_deep' => ['T', '@phpstan-template'=>'T', 'value'=>'T'],
-    'urldecode_deep' => ['T', '@phpstan-template'=>'T', 'value'=>'T'],
-    'urlencode_deep' => ['T', '@phpstan-template'=>'T', 'value'=>'T'],
-    'wp_clear_scheduled_hook' => ['($wp_error is false ? 0|positive-int|false : 0|positive-int|\WP_Error)', 'args'=>$cronArgsType],
+    'addslashes_gpc' => ['T', '@phpstan-template' => 'T', 'gpc' => 'T'],
+    'get_objects_in_term' => [null, 'args' => 'array{order?: string}'],
+    'have_posts' => [null, '@phpstan-impure' => ''],
+    'rawurlencode_deep' => ['T', '@phpstan-template' => 'T', 'value' => 'T'],
+    'sanitize_category' => ['T', '@phpstan-template' => 'T of array|object', 'category' => 'T'],
+    'sanitize_post' => ['T', '@phpstan-template' => 'T of array|object', 'post' => 'T'],
+    'sanitize_term' => ['T', '@phpstan-template' => 'T of array|object', 'term' => 'T'],
+    'stripslashes_deep' => ['T', '@phpstan-template' => 'T', 'value' => 'T'],
+    'urldecode_deep' => ['T', '@phpstan-template' => 'T', 'value' => 'T'],
+    'urlencode_deep' => ['T', '@phpstan-template' => 'T', 'value' => 'T'],
+    'wp_clear_scheduled_hook' => ['(0|positive-int|($wp_error is false ? false : \WP_Error))', 'args'=>$cronArgsType],
```

Comment:
I wasn't aware PHPStan supported this syntax for conditional return types, nice

Reply:
Me neither. It's great! It helps to simplify some of the conditions and make them easier to read.